addons/website/static/src/snippets/s_embed_code/options.js

```diff
@@ -12,8 +12,31 @@ options.registry.EmbedCode = options.Class.extend({
     //--------------------------------------------------------------------------
 
     async editCode() {
+        await new Promise((resolve, reject) => {
+            const dialog = new Dialog(this, {
+                title: _t("Do not copy/paste code you do not understand, this could put your data at risk."),
+                size: "medium",
+                $content: $("<div/>", {text: _t("If you need to add analytics or marketing tags, inject code in your <head> or <body> instead. The option is in the \"Theme\" tab.")}),
+                buttons: [
+                    {
+                        text: _t("OK"),
+                        classes: "btn-primary",
+                        click: () => resolve(),
+                        close: true,
+                    },
+                    {
+                        text: _t("Discard"),
+                        click: () => reject(),
+                        close: true,
+                    },
+                ],
+            });
+            dialog.on("closed", this, reject);
+            dialog.open();
+        });
+
         const $container = this.$target.find('.s_embed_code_embedded');
-        const code = $container.html().trim();
+        let embedContent = $container[0].dataset.embedContent;
```

Comment:
I don't think you are supporting previously added embed code, should you not fallback on reading the html content if no embedContent attribute is there? + add a comment?